### PR TITLE
[ES-1043] added v2/kyc-exchange endpoint

### PR DESCRIPTION
**Pull request by kaifk468**

### CODE REVIEW

File: mock-identity-system/src/main/java/io/mosip/esignet/mock/identitysystem/controller/AuthController.java [hunk 1]
```diff
@@ -38,6 +39,20 @@ public ResponseWrapper<KycExchangeResponseDto> kycExchange(@PathVariable @NotBla
...
+    @PostMapping(path = "v2/kyc-exchange/{relyingPartyId}/{clientId}",
+            consumes = MediaType.APPLICATION_JSON_VALUE, produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseWrapper<KycExchangeResponseDto> kycExchangeV2(@PathVariable @NotBlank String relyingPartyId,
+                                                               @PathVariable @NotBlank String clientId,
+                                                               @RequestBody @NotNull @Valid KycExchangeRequestDtoV2 kycExchangeRequestDtoV2) {
+        ResponseWrapper<KycExchangeResponseDto> responseWrapper = new ResponseWrapper<>();
+        try {
+            responseWrapper.setResponse(authenticationService.kycExchangeV2(relyingPartyId, clientId, kycExchangeRequestDtoV2));
+        } catch (MockIdentityException ex) {
+            throw ex;
```
Comment (ase-101) [inline mock-identity-system/src/main/java/io/mosip/esignet/mock/identitysystem/controller/AuthController.java:51]: You should throw KYCExchaneException not the MockIdentityException

File: mock-identity-system/src/main/java/io/mosip/esignet/mock/identitysystem/service/IdentityService.java [hunk 2]
```diff
@@ -11,4 +17,6 @@ public interface IdentityService {
 	public IdentityData getIdentity(String individualId) throws MockIdentityException;
 
     public  void addVerifiedClaim(VerifiedClaimRequestDto request) throws MockIdentityException;
+
+	public JsonNode getIdentityInJsonNode(String individualId) throws MockIdentityException;
```
Comment (ase-101) [inline mock-identity-system/src/main/java/io/mosip/esignet/mock/identitysystem/service/IdentityService.java:21]: Kindly rename the method

File: mock-identity-system/src/main/java/io/mosip/esignet/mock/identitysystem/service/impl/AuthenticationServiceImpl.java [hunk 3]
```diff
@@ -257,7 +301,7 @@ private Boolean doKycAuthentication(KycAuthRequestDto kycAuthRequestDto,Identity
         }
 
         if (kycAuthRequestDto.getPin() != null) {
-            authStatus = kycAuthRequestDto.getPin().equals(identityData.getPin());
+            authStatus = kycAuthRequestDto.getPin().equals(HelperUtil.getIdentityDataValue(identityData,"pin","eng"));
```
Comment (ase-101) [inline mock-identity-system/src/main/java/io/mosip/esignet/mock/identitysystem/service/impl/AuthenticationServiceImpl.java:304]: lets use already defined "defaultLanguage" variable.

Comment (kaifk468) [reply mock-identity-system/src/main/java/io/mosip/esignet/mock/identitysystem/service/impl/AuthenticationServiceImpl.java:304]: used the default language

File: mock-identity-system/src/main/java/io/mosip/esignet/mock/identitysystem/service/IdentityService.java (hunk 2 above)
Comment (kaifk468) [reply mock-identity-system/src/main/java/io/mosip/esignet/mock/identitysystem/service/IdentityService.java:21]: changed it to getIdentityV2 as method overloading is not possible with different return type

**Review by ase-101 — APPROVED** [review]

**Pull request merged**